### PR TITLE
feat(url): add sub-domains handling in url construction

### PR DESCRIPTION
**Pull request by zarov**
Solves #942

### CODE REVIEW

File: src/Provider/URLBuilder.js [hunk 1]
```diff
@@ -1,14 +1,37 @@
 import Extent from 'Core/Geographic/Extent';
 
 const extent = new Extent('EPSG:4326', [0, 0, 0, 0]);
+
+let subDomainsCount = 0;
+function subDomains(url) {
+    var subDomainsPtrn = /\$\{u:([\w-_.|]+)\}/.exec(url);
```
Comment (gchoqueux) [inline src/Provider/URLBuilder.js]: could use const?

File: src/Provider/URLBuilder.js [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+
+    if (!subDomainsPtrn) {
+        return url;
+    }
+
+    var subDomainsList = subDomainsPtrn[1].split('|');
```
Comment (gchoqueux) [inline src/Provider/URLBuilder.js]: could use const?

**Review by gchoqueux — APPROVED** [review]

File: src/Provider/URLBuilder.js (hunk 1 above)
Comment (zarov) [reply src/Provider/URLBuilder.js]: done

File: src/Provider/URLBuilder.js (hunk 2 above)
Comment (zarov) [reply src/Provider/URLBuilder.js]: done

**Pull request merged**